### PR TITLE
Run missing tests

### PR DESCRIPTION
**Pull request by jclarkeSTFC**
In [this](https://github.com/mantidproject/mantid/commit/bccb6263b4538be1191c30afe80c55c91b648431) commit, `${GMOCK_TEST_FILES}` went missing when the tests are added, which means `ImplicitFunctionFactoryTest`, `ImplicitFunctionParameterParserFactoryTest`, and `MatrixWorkspaceTest` haven't been run for 3 years.

One of the tests in `MatrixWorkspaceTest` spotted a (niche) bug, which is also fixed in this PR. If there are no times in your sample log after 1991, then instead of returning the last time, it was indexing the memory outside the range of the time vector. 

### To test:

The tests mentioned above should run.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Framework/API/src/LogManager.cpp [hunk 1]
```diff
@@ -212,17 +212,13 @@ const DateAndTime LogManager::getFirstPulseTime() const {
   // a DAS bug at SNS around Mar 2011 where the first pulse time is Jan 1, 1990."
   // There was no explanation why 100 was picked as the maximum number of times to skip.
```
Comment (cailafinn) [inline Framework/API/src/LogManager.cpp:213]: If we're no longer limiting to 100 we should probably update the comment to match.

**Review by cailafinn — CHANGES_REQUESTED** [review]
Functionally working well, test is now running properly locally and on CI. Nice find on the other bugfix, just a request to update the comment to match the slightly changed functionality.

File: Framework/API/src/LogManager.cpp (hunk 1 above)
Comment (jclarkeSTFC) [reply Framework/API/src/LogManager.cpp:213]: I've changed it so the 100 limit is still there, just in case

**Review by cailafinn — APPROVED** [review]
Code changes look good, working well as test now running. Nice work :shipit:

**Pull request merged**